src/build/app/mod.rs

```diff
@@ -1461,6 +1461,36 @@ impl<'b> App<'b> {
         debug_assert!(self._app_debug_asserts());
         self.args._build();
         self.settings.set(AppSettings::Built);
+        Self::_panic_on_missing_help(&self, self.g_settings.is_set(AppSettings::HelpRequired));
```

Comment:
I am not sure if you need the second argument. I think global settings should propagate completely.

Reply:
I thought so too.... But see  my [previous comment]( https://github.com/clap-rs/clap/pull/1683#issuecomment-584126466)

Maybe there is a better way?

Reply:
Hmm.. So, I was looking at the code and it looks `App._propagate` happens during parsing.

I am merging this PR for now. But I think we should refactor such that `_propagate` happens during `_build`. It looks it have been done this way to reduce running times. @CreepySkeleton could you look at the code once?

Reply:
Will do, but not now and not tomorrow, maybe the day after it. Tomorrow I'm working on fixing my "extract subcommands" PR.